File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.py
class Solution:
    def areOccurrencesEqual(self, s: str) -> bool:
        mp = {}
        for c in s:
            mp[c] = mp.get(c, 0) + 1
        cnt = None
        for k in mp:
            if not cnt:
                cnt = mp[k]
            elif mp[k] != cnt:
                return False
        return True
            

File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.js
/**
 * @param {string} s
 * @return {boolean}
 */
var areOccurrencesEqual = function(s) {
    const map = {}
    for (const c of s) {
        map[c] = map[c] + 1 || 1
    }
    let cnt
    for (const key in map) {
        if (!cnt) {
            cnt = map[key]
        } else if (map[key] !== cnt) {
            return false
        }
    }
    return true
};

areOccurrencesEqual = function(s) {
    return (new Set(Object.values(s.split("").reduce((acc, val) => ({...acc, [val]: acc[val] + 1 || 1}), {})))).size === 1
}